packages/accounts-passwordless/passwordless_server.js

```diff
@@ -12,9 +12,19 @@ const findUserWithOptions = ({ selector }) => {
   if (!selector) {
     Accounts._handleError('A selector is necessary');
   }
-  const { email, ...rest } = selector;
+  const { email, id, ...rest } = selector;
+
+  const modifiedSelector = { ...rest };
+  if (id) {
+    modifiedSelector._id = id;
+  } 
+
+   if (email) {
+    modifiedSelector['emails.address'] = email;
+  }
+
   return Meteor.users.findOne(
-    { ...rest, ...(email ? { 'emails.address': selector.email } : {}) },
+    modifiedSelector,
```

Comment:
then create the selector directly here.

Reply:
Nice, If I understood, something like that?
```js
const findUserWithOptions = ({ selector }) => {
  if (!selector) {
    Accounts._handleError('A selector is necessary');
  }
  const { email, id, ...rest } = selector;

  return Meteor.users.findOne(
    { 
      ...rest,
      ...(id && {_id: id}), 
      ...(email && {'emails.address': email}) 
    },
    {
      fields: {
        services: 1,
        emails: 1,
      },
    }
  );
};
```

With it, it is the same logic, but without mutations, I do not need to create a new `const`.